### ISSUE TITLE
BUG: `GanttSeries.from_pandas()` now fails on inherited validation

### ISSUE THREAD

**Issue by hcpchris** [highcharts-for-python/highcharts-gantt#46]
Following the upgrade to v.1.4, the ``GantSeries.from_pandas()`` method is now breaking when evaluating the truthiness of a NumPy array as outlined in [this forum post](https://www.highcharts.com/forum/posting.php?mode=reply&f=19&t=51825#pr190358).

**Comment by hcpchris** [comment]
This issue really needs to be addressed in Highcharts Core for Python. I've filed a separate bug report there: highcharts-for-python/highcharts-core#162.

**Comment by hcpchris** [comment]
Turns out that it was an error partially in Highcharts Gantt for Python and partially in Highcharts Core for Python. The root cause of the problem was that the ``GanttSeries._data_collection_class()`` method was actually returning the data *point* class, while the ``._data_point_class()`` method was returning the *collection* class (they were swapped). 

Fixing this issue actually uncovered a separate problem, namely that date validation fails when converting between ``numpy.datetime64`` and ``datetime.datetime``. I implemented ``utility_functions.datetime64_to_datetime`` in the Highcharts Core for Python library, until such time as the Validator Collection extends datetime validation coverage to support ``numpy.datetime64``.
